Subscribers: fix time since

arcangelini opened the pull request:
## Proposed Changes

Switch from `Since` subscribed to `Date Subscribed` in subscribers data view.

<img width="1122" alt="Screen Shot 2025-05-29 at 16 43 29" src="https://github.com/user-attachments/assets/e70cd441-fbb8-469e-be97-97e1176b211b" />

## Why are these changes being made?

Currently the time difference from subscribed is checking against UTC time rather than the sites time. Adding the `+00:00` to the subscribed date fixes that part.

I also don't think it is necessary to show "time since" in this area. There doesn't appear to be a reason that someone would need this detail over just the subscribed date. Moving this to just a simple date simplifies things I believe.

## Testing Instructions

1. Pull the branch and navigate to `/subscribers/YOUR_SITE.com`
2. Check that the Date Subscribed column is showing the correct dates in the correct order.
3. Add a new subscriber and verify the order is still correct.
4. Change timezone on your site and make sure the order is still correct.
5. Change languages for your user and ensure it shows the proper date.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

arcangelini committed:
Subscribers: fix time since

arcangelini committed:
Refactor and fix bug

arcangelini committed:
Pivot and change to Date Subscribed

Addison-Stavlo commented on client/my-sites/subscribers/components/subscriber-data-views/subscriber-data-views.tsx:
```diff
@@ -368,11 +374,10 @@ export default function SubscriberDataViews( {
-				render: ( { item }: { item: Subscriber } ) => (
-					<TimeSince date={ getSubscriptionDate( item ) } />
-				),
+				label: translate( 'Date Subscribed' ),
```
I think technically we do all labels in sentence case now, although this is currently all caps in the interface here anyways. It may be good to move it to sentence case for consistency if the typography styles were changed.

It also doesn't look like we have existing translations for "Date Subscribed". So we could use the `fixMe` function from the `i18n-calypso` library to provide this new translation with the old translation as a fallback while the new one is being translated.

arcangelini replied:
Great call! Thanks @Addison-Stavlo

Addison-Stavlo reviewed (approved):
This tests well on my end. I just left some minor notes on the new translation.

arcangelini committed:
Subscribers: Use fixMe for 'Date subscribed' label translation

Update the date_subscribed field label to use fixMe() to handle the copy change from 'Since' to 'Date subscribed'. Also added locale to the useMemo dependency array.

arcangelini committed:
Merge branch 'trunk' into fix/issue-cml-530
